tools/wake-format/emitter.cpp

```diff
@@ -403,7 +436,66 @@ wcl::optional<wcl::doc> Emitter::walk_data(ctx_t ctx, CSTElement node) {
 }
 
 wcl::optional<wcl::doc> Emitter::walk_def(ctx_t ctx, CSTElement node) {
-  return walk_placeholder(ctx, node);
+  ASSERT_TOKEN(node, CST_DEF);
+  wcl::doc_builder bdr;
+  CSTElement child = node.firstChildElement();
+
+  // def
+  ASSERT_TOKEN(child, TOKEN_KW_DEF);
+  bdr.append("def");
+  child.nextSiblingElement();
+
+  // ws
+  ASSERT_TOKEN(child, TOKEN_WS);
+  bdr.append(space());
+  child.nextSiblingElement();
+
+  // CST_ID, CST_APPLY, TODO: others?
```

Comment:
Not requesting a change just giving info:

string and integer literals are the others but you can do those later. `def 1 = 2` parses, it just fails to pass semantic checks because `1` is not an exhaustive pattern. You can also have parens as well in patterns which I think you need to handle eventully.

Also if binary/unary ops aren't handled by CST_APPLY then you need to handle those as well.

Another thing to check is that `def (if x then y else z) = 10` is a *parse* error and not just a semantic error, otherwise you actually have to handle any arbitrary expression

Reply:
yeah, makes sense! Thanks for the context